[Ng Yin Joe] iP

joeng03 opened the pull request:
### Linus 

> "The common man is not concerned about the passage of time, the man of talent is driven by it." - Arthur Schopenhauer ([source](https://quotefancy.com/quote/768387/Arthur-Schopenhauer-The-common-man-is-not-concerned-about-the-passage-of-time-the-man-of))

Linus frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. Download it from [here](https://github.com/joeng03/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Type of tasks:

- ToDo
- Deadline 
- Event

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

joeng03 committed:
feat: rename, greet, exit

joeng03 committed:
feat: Echo

joeng03 committed:
feat: Add, List

joeng03 committed:
feat: Mark as Done

joeng03 committed:
feat: ToDos, Events, Deadlines

joeng03 committed:
test: Automated Text UI Testing

joeng03 committed:
feat: Handle Errors

joeng03 committed:
feat: Delete

joeng03 committed:
refactor: Use Enums

joeng03 committed:
style: Minor adjustmets to comply with the Java coding standard

joeng03 committed:
refactor: Rename MessagePrint to MessagePrinter

joeng03 committed:
feat: Integrate Dates and Times into Deadline and Event

joeng03 committed:
Merge branch 'branch-Level-8'

joeng03 committed:
Merge branch 'add-gradle-support' of https://github.com/joeng03/ip

joeng03 committed:
feat: Add save tasks functionality

joeng03 committed:
Merge branch 'branch-Level-7'

joeng03 committed:
refactor: Use OOP to refactor the codebase

joeng03 committed:
Merge branch 'A-MoreOOP'

joeng03 committed:
refactor: Organize classes into packages

joeng03 committed:
fix: Gson LocalDate serialization issues

joeng03 committed:
merge branch A-Packages

joeng03 committed:
refactor: Add parent package linus

joeng03 committed:
test: Add JUnit tests

joeng03 committed:
Merge branch 'A-JUnit'

joeng03 committed:
docs: Add JavaDoc documentation

joeng03 committed:
style: Modify code to adhere to coding standard

joeng03 committed:
feat: Add find command and functionality

joeng03 committed:
Merge branch 'A-JavaDoc'

joeng03 committed:
merge A-CodingStandard

joeng03 committed:
Merge branch 'Level-9'

joeng03 committed:
style: Add and follow checkstyle

joeng03 committed:
merge changes from A-CheckStyle

joeng03 committed:
fix: Checkstyle error for find() method due to absence of JavaDoc

joeng03 committed:
fix: Sync the taskList to disk everytime the taskList changes

joeng03 committed:
fix: index off-by-one error for delete, mark, and unmark

joeng03 committed:
feat: Add GUI with JavaFX

joeng03 committed:
merge changes from branch-Level-10

joeng03 committed:
refactor: Reorganize folders

chew01 commented on data/linus.txt:
```diff
@@ -0,0 +1 @@
+[{"type":"linus.task.Deadline","by":"2023-08-31","description":"task 2","isDone":false},{"type":"linus.task.Deadline","by":"2022-01-01","description":"TEST","isDone":true},{"type":"linus.task.ToDo","description":"ggg","isDone":true},{"type":"linus.task.ToDo","description":"johjh","isDone":true},{"type":"linus.task.Deadline","by":"2100-03-02","description":"TEST","isDone":true},{"type":"linus.task ...
```
This might be beyond the scope of the review, but perhaps the data folder should not be tracked in git?

chew01 commented on src/main/java/linus/Linus.java:
```diff
@@ -0,0 +1,155 @@
+     * @param filePath The file path.
+     */
+    public Linus(String filePath) {
+        storage = new Storage("data/linus.txt");
```
As you have a constant FILE_PATH above, perhaps it's a good idea to use it here?

chew01 commented on src/main/java/linus/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+ */
+public class Deadline extends Task {
+    protected String type = "linus.task.Deadline";
+    protected LocalDate by = null;
```
Perhaps the name of the `by` variable could be more detailed?

chew01 commented on src/main/java/linus/task/Event.java:
```diff
@@ -0,0 +1,67 @@
+ */
+public class Event extends Task {
+    protected String type = "linus.task.Event";
+    protected LocalDate from = null;
+    protected LocalDate to = null;
```
As above, perhaps these variables could be more detailed?

chew01 reviewed (commented):
Your code is cool 😎 I like the idea of using JSON format to store data!
Just a few nits about variable naming.

joeng03 committed:
chore: Cleanup redundant code

joeng03 committed:
chore: Cleanup System.out.println

joeng03 committed:
test: Add assertions

joeng03 committed:
Merge pull request #1 from joeng03/A-Assertions

test: Add assertions

joeng03 committed:
Merge pull request #1 from joeng03/A-Assertions

test: Add assertions

joeng03 committed:
Merge branch 'master' into A-Assertions

joeng03 committed:
refactor: Improve Code quality

joeng03 committed:
Merge pull request #2 from joeng03/A-CodeQuality

refactor: Improve Code quality

joeng03 committed:
Merge branch 'master' of https://github.com/joeng03/ip

joeng03 committed:
chore: Add CI workflow

joeng03 committed:
fix: Checkstyle errors

joeng03 committed:
fix: Remove IO redirection test

joeng03 committed:
refactor: organize messages as constants

joeng03 committed:
feat:Add exit functionality after user enters bye command

If the user enters the bye command,
the MainWindow controller detects it,
then responds with the exit message.
It then uses JavaFX PauseTransition
to wait for 1.5 seconds before exiting.

This ensures that the window exits seamlessly
while also improving user experience,
as users could read the exit message
before the window exits.

joeng03 committed:
feat:add exit functionality after user types bye command

joeng03 committed:
Merge branch 'master' of https://github.com/joeng03/ip

joeng03 committed:
feat: Add statistics functionality for tasks

joeng03 committed:
Merge branch 'BCD-Extension'

joeng03 committed:
test: Add new test cases for tasklist

joeng03 committed:
refactor: Abstract commands away

joeng03 committed:
docs: Add User Guide

joeng03 committed:
docs: Add Product Screenshot

joeng03 committed:
docs: Fix typos

joeng03 committed:
Merge branch 'branch-A-Assertions'

joeng03 committed:
fix: Create file when it does not exist yet

joeng03 committed:
fix: Remove print statements
